bug(gen): Regular expressions with Unicode symbols panic on application startup.

<!-- Please answer these questions before submitting your issue. Thanks! -->

### What version of ogen are you using?

```bash
$ go list -m github.com/ogen-go/ogen
github.com/ogen-go/ogen v1.8.1
```

### Can this issue be reproduced with the latest version?

Yes

### What did you do?

TLDR: Patterns with unicode symbols are invalid after several compilations from ECMA to golang regexp specification.
<!--
Write the steps for reproducing the bug.

Link to schema/small snippet of schema will be helpful.
-->
Swagger code:
```yml
parameters:
  Name:
    name: name
    in: query
    schema:
      type: string
      pattern: '[\u2000-\u206F\u2E00-\u2E7F]'
```
Generated code (oas_cfg_gen):
```go
var regexMap = map[string]ogenregex.Regexp{
	"[\\x{2000}-\\x{206F}\\x{2E00}-\\x{2E7F}]": ogenregex.MustCompile("[\\x{2000}-\\x{206F}\\x{2E00}-\\x{2E7F}]"),
}
```

### What did you expect to see?

An application starts successfully.

The generated code already has a valid regexp and it should just use the `regexp` package.

### What did you see instead?

The second ECMA to go regexp compilation broke regex itself.

```sh
panic: regexp2: error parsing regexp: [}-x] range in reverse order in `[\x{2000}-\x{206F}\x{2E00}-\x{2E7F}]`

goroutine 1 [running]:
github.com/go-faster/errors.Must[...](...)
        /Users/user/go/pkg/mod/github.com/go-faster/errors@v0.7.1/must.go:9
github.com/ogen-go/ogen/ogenregex.MustCompile({0x102f38dc5?, 0x102ef5d98?})
        /Users/user/go/pkg/mod/github.com/ogen-go/ogen@v1.8.1/ogenregex/ogenregex.go:88 +0x44
ogen-test/pkg/gen.init()
        /Users/user/Desktop/git/ogen-test/pkg/gen/oas_cfg_gen.go:20 +0x30
exit status 2

```

### Related links
Possible reason: [PR!641](https://github.com/ogen-go/ogen/pull/641).

--- a/internal/integration/sample_api/oas_cfg_gen.go
+++ b/internal/integration/sample_api/oas_cfg_gen.go
@@ -19,10 +19,10 @@ import (
 )
 
 var regexMap = map[string]ogenregex.Regexp{
-	"^\\d-\\d$":                      ogenregex.MustCompile("^\\d-\\d$"),
-	"^variant3_[^\r\n\u2028\u2029]*": ogenregex.MustCompile("^variant3_[^\r\n\u2028\u2029]*"),
-	"foo[^\r\n\u2028\u2029]*":        ogenregex.MustCompile("foo[^\r\n\u2028\u2029]*"),
-	"string_[^\r\n\u2028\u2029]*":    ogenregex.MustCompile("string_[^\r\n\u2028\u2029]*"),
+	"^\\d-\\d$":    ogenregex.MustCompile("^\\d-\\d$"),
+	"^variant3_.*": ogenregex.MustCompile("^variant3_.*"),
+	"foo.*":        ogenregex.MustCompile("foo.*"),
+	"string_.*":    ogenregex.MustCompile("string_.*"),
 }
 var ratMap = map[string]*big.Rat{
 	"10": func() *big.Rat {

--- a/internal/integration/sample_api/oas_json_gen.go
+++ b/internal/integration/sample_api/oas_json_gen.go
@@ -2942,7 +2942,7 @@ func (s *Issue943Map) Decode(d *jx.Decoder) error {
 			}
 		default:
 			var handled bool
-			switch match, err := regexMap["^variant3_[^\r\n\u2028\u2029]*"].Match(k); {
+			switch match, err := regexMap["^variant3_.*"].Match(k); {
 			case err != nil:
 				return errors.Wrap(err, "execute regex")
 			case match:
@@ -3040,7 +3040,7 @@ func (s *Issue943MapPattern0) Decode(d *jx.Decoder) error {
 		return errors.New("invalid: unable to decode Issue943MapPattern0 to nil")
 	}
 	m := s.init()
-	pattern := regexMap["^variant3_[^\r\n\u2028\u2029]*"]
+	pattern := regexMap["^variant3_.*"]
 	if err := d.ObjBytes(func(d *jx.Decoder, k []byte) error {
 		switch match, err := pattern.Match(k); {
 		case err != nil:
@@ -6092,7 +6092,7 @@ func (s *OnlyPatternedPropsObject) Decode(d *jx.Decoder) error {
 		return errors.New("invalid: unable to decode OnlyPatternedPropsObject to nil")
 	}
 	m := s.init()
-	pattern := regexMap["string_[^\r\n\u2028\u2029]*"]
+	pattern := regexMap["string_.*"]
 	if err := d.ObjBytes(func(d *jx.Decoder, k []byte) error {
 		switch match, err := pattern.Match(k); {
 		case err != nil:
@@ -7625,7 +7625,7 @@ func (s *PatternRecursiveMap) Decode(d *jx.Decoder) error {
 		return errors.New("invalid: unable to decode PatternRecursiveMap to nil")
 	}
 	m := s.init()
-	pattern := regexMap["foo[^\r\n\u2028\u2029]*"]
+	pattern := regexMap["foo.*"]
 	if err := d.ObjBytes(func(d *jx.Decoder, k []byte) error {
 		switch match, err := pattern.Match(k); {
 		case err != nil:
@@ -8742,7 +8742,7 @@ func (s *StringIntMap) Decode(d *jx.Decoder) error {
 		switch string(k) {
 		default:
 			var handled bool
-			switch match, err := regexMap["string_[^\r\n\u2028\u2029]*"].Match(k); {
+			switch match, err := regexMap["string_.*"].Match(k); {
 			case err != nil:
 				return errors.Wrap(err, "execute regex")
 			case match:
@@ -8875,7 +8875,7 @@ func (s *StringIntMapPattern0) Decode(d *jx.Decoder) error {
 		return errors.New("invalid: unable to decode StringIntMapPattern0 to nil")
 	}
 	m := s.init()
-	pattern := regexMap["string_[^\r\n\u2028\u2029]*"]
+	pattern := regexMap["string_.*"]
 	if err := d.ObjBytes(func(d *jx.Decoder, k []byte) error {
 		switch match, err := pattern.Match(k); {
 		case err != nil:

--- a/internal/integration/sample_api/oas_schemas_gen.go
+++ b/internal/integration/sample_api/oas_schemas_gen.go
@@ -1311,7 +1311,7 @@ func NewIssue943MapIssue943(v Issue943Map) Issue943 {
 // Ref: #/components/schemas/Issue943Map
 type Issue943Map struct {
 	Selector string `json:"selector"`
-	// Pattern: "^variant3_[^\r\n\u2028\u2029]*".
+	// Pattern: "^variant3_.*".
 	Pattern0Props Issue943MapPattern0
 }
 
@@ -5544,7 +5544,7 @@ func (s *RecursiveMapAdditional) init() RecursiveMapAdditional {
 // Ref: #/components/schemas/StringIntMap
 type StringIntMap struct {
 	AdditionalProps StringIntMapAdditional
-	// Pattern: "string_[^\r\n\u2028\u2029]*".
+	// Pattern: "string_.*".
 	Pattern0Props StringIntMapPattern0
 }
 

--- a/internal/integration/sample_api_nc/oas_cfg_gen.go
+++ b/internal/integration/sample_api_nc/oas_cfg_gen.go
@@ -18,10 +18,10 @@ import (
 )
 
 var regexMap = map[string]ogenregex.Regexp{
-	"^\\d-\\d$":                      ogenregex.MustCompile("^\\d-\\d$"),
-	"^variant3_[^\r\n\u2028\u2029]*": ogenregex.MustCompile("^variant3_[^\r\n\u2028\u2029]*"),
-	"foo[^\r\n\u2028\u2029]*":        ogenregex.MustCompile("foo[^\r\n\u2028\u2029]*"),
-	"string_[^\r\n\u2028\u2029]*":    ogenregex.MustCompile("string_[^\r\n\u2028\u2029]*"),
+	"^\\d-\\d$":    ogenregex.MustCompile("^\\d-\\d$"),
+	"^variant3_.*": ogenregex.MustCompile("^variant3_.*"),
+	"foo.*":        ogenregex.MustCompile("foo.*"),
+	"string_.*":    ogenregex.MustCompile("string_.*"),
 }
 var ratMap = map[string]*big.Rat{
 	"10": func() *big.Rat {

--- a/internal/integration/sample_api_nc/oas_json_gen.go
+++ b/internal/integration/sample_api_nc/oas_json_gen.go
@@ -2942,7 +2942,7 @@ func (s *Issue943Map) Decode(d *jx.Decoder) error {
 			}
 		default:
 			var handled bool
-			switch match, err := regexMap["^variant3_[^\r\n\u2028\u2029]*"].Match(k); {
+			switch match, err := regexMap["^variant3_.*"].Match(k); {
 			case err != nil:
 				return errors.Wrap(err, "execute regex")
 			case match:
@@ -3040,7 +3040,7 @@ func (s *Issue943MapPattern0) Decode(d *jx.Decoder) error {
 		return errors.New("invalid: unable to decode Issue943MapPattern0 to nil")
 	}
 	m := s.init()
-	pattern := regexMap["^variant3_[^\r\n\u2028\u2029]*"]
+	pattern := regexMap["^variant3_.*"]
 	if err := d.ObjBytes(func(d *jx.Decoder, k []byte) error {
 		switch match, err := pattern.Match(k); {
 		case err != nil:
@@ -6092,7 +6092,7 @@ func (s *OnlyPatternedPropsObject) Decode(d *jx.Decoder) error {
 		return errors.New("invalid: unable to decode OnlyPatternedPropsObject to nil")
 	}
 	m := s.init()
-	pattern := regexMap["string_[^\r\n\u2028\u2029]*"]
+	pattern := regexMap["string_.*"]
 	if err := d.ObjBytes(func(d *jx.Decoder, k []byte) error {
 		switch match, err := pattern.Match(k); {
 		case err != nil:
@@ -7625,7 +7625,7 @@ func (s *PatternRecursiveMap) Decode(d *jx.Decoder) error {
 		return errors.New("invalid: unable to decode PatternRecursiveMap to nil")
 	}
 	m := s.init()
-	pattern := regexMap["foo[^\r\n\u2028\u2029]*"]
+	pattern := regexMap["foo.*"]
 	if err := d.ObjBytes(func(d *jx.Decoder, k []byte) error {
 		switch match, err := pattern.Match(k); {
 		case err != nil:
@@ -8742,7 +8742,7 @@ func (s *StringIntMap) Decode(d *jx.Decoder) error {
 		switch string(k) {
 		default:
 			var handled bool
-			switch match, err := regexMap["string_[^\r\n\u2028\u2029]*"].Match(k); {
+			switch match, err := regexMap["string_.*"].Match(k); {
 			case err != nil:
 				return errors.Wrap(err, "execute regex")
 			case match:
@@ -8875,7 +8875,7 @@ func (s *StringIntMapPattern0) Decode(d *jx.Decoder) error {
 		return errors.New("invalid: unable to decode StringIntMapPattern0 to nil")
 	}
 	m := s.init()
-	pattern := regexMap["string_[^\r\n\u2028\u2029]*"]
+	pattern := regexMap["string_.*"]
 	if err := d.ObjBytes(func(d *jx.Decoder, k []byte) error {
 		switch match, err := pattern.Match(k); {
 		case err != nil:

--- a/internal/integration/sample_api_nc/oas_schemas_gen.go
+++ b/internal/integration/sample_api_nc/oas_schemas_gen.go
@@ -1311,7 +1311,7 @@ func NewIssue943MapIssue943(v Issue943Map) Issue943 {
 // Ref: #/components/schemas/Issue943Map
 type Issue943Map struct {
 	Selector string `json:"selector"`
-	// Pattern: "^variant3_[^\r\n\u2028\u2029]*".
+	// Pattern: "^variant3_.*".
 	Pattern0Props Issue943MapPattern0
 }
 
@@ -5544,7 +5544,7 @@ func (s *RecursiveMapAdditional) init() RecursiveMapAdditional {
 // Ref: #/components/schemas/StringIntMap
 type StringIntMap struct {
 	AdditionalProps StringIntMapAdditional
-	// Pattern: "string_[^\r\n\u2028\u2029]*".
+	// Pattern: "string_.*".
 	Pattern0Props StringIntMapPattern0
 }
 

--- a/internal/integration/sample_api_no_otel/oas_cfg_gen.go
+++ b/internal/integration/sample_api_no_otel/oas_cfg_gen.go
@@ -14,10 +14,10 @@ import (
 )
 
 var regexMap = map[string]ogenregex.Regexp{
-	"^\\d-\\d$":                      ogenregex.MustCompile("^\\d-\\d$"),
-	"^variant3_[^\r\n\u2028\u2029]*": ogenregex.MustCompile("^variant3_[^\r\n\u2028\u2029]*"),
-	"foo[^\r\n\u2028\u2029]*":        ogenregex.MustCompile("foo[^\r\n\u2028\u2029]*"),
-	"string_[^\r\n\u2028\u2029]*":    ogenregex.MustCompile("string_[^\r\n\u2028\u2029]*"),
+	"^\\d-\\d$":    ogenregex.MustCompile("^\\d-\\d$"),
+	"^variant3_.*": ogenregex.MustCompile("^variant3_.*"),
+	"foo.*":        ogenregex.MustCompile("foo.*"),
+	"string_.*":    ogenregex.MustCompile("string_.*"),
 }
 var ratMap = map[string]*big.Rat{
 	"10": func() *big.Rat {

--- a/internal/integration/sample_api_no_otel/oas_json_gen.go
+++ b/internal/integration/sample_api_no_otel/oas_json_gen.go
@@ -2942,7 +2942,7 @@ func (s *Issue943Map) Decode(d *jx.Decoder) error {
 			}
 		default:
 			var handled bool
-			switch match, err := regexMap["^variant3_[^\r\n\u2028\u2029]*"].Match(k); {
+			switch match, err := regexMap["^variant3_.*"].Match(k); {
 			case err != nil:
 				return errors.Wrap(err, "execute regex")
 			case match:
@@ -3040,7 +3040,7 @@ func (s *Issue943MapPattern0) Decode(d *jx.Decoder) error {
 		return errors.New("invalid: unable to decode Issue943MapPattern0 to nil")
 	}
 	m := s.init()
-	pattern := regexMap["^variant3_[^\r\n\u2028\u2029]*"]
+	pattern := regexMap["^variant3_.*"]
 	if err := d.ObjBytes(func(d *jx.Decoder, k []byte) error {
 		switch match, err := pattern.Match(k); {
 		case err != nil:
@@ -6092,7 +6092,7 @@ func (s *OnlyPatternedPropsObject) Decode(d *jx.Decoder) error {
 		return errors.New("invalid: unable to decode OnlyPatternedPropsObject to nil")
 	}
 	m := s.init()
-	pattern := regexMap["string_[^\r\n\u2028\u2029]*"]
+	pattern := regexMap["string_.*"]
 	if err := d.ObjBytes(func(d *jx.Decoder, k []byte) error {
 		switch match, err := pattern.Match(k); {
 		case err != nil:
@@ -7625,7 +7625,7 @@ func (s *PatternRecursiveMap) Decode(d *jx.Decoder) error {
 		return errors.New("invalid: unable to decode PatternRecursiveMap to nil")
 	}
 	m := s.init()
-	pattern := regexMap["foo[^\r\n\u2028\u2029]*"]
+	pattern := regexMap["foo.*"]
 	if err := d.ObjBytes(func(d *jx.Decoder, k []byte) error {
 		switch match, err := pattern.Match(k); {
 		case err != nil:
@@ -8742,7 +8742,7 @@ func (s *StringIntMap) Decode(d *jx.Decoder) error {
 		switch string(k) {
 		default:
 			var handled bool
-			switch match, err := regexMap["string_[^\r\n\u2028\u2029]*"].Match(k); {
+			switch match, err := regexMap["string_.*"].Match(k); {
 			case err != nil:
 				return errors.Wrap(err, "execute regex")
 			case match:
@@ -8875,7 +8875,7 @@ func (s *StringIntMapPattern0) Decode(d *jx.Decoder) error {
 		return errors.New("invalid: unable to decode StringIntMapPattern0 to nil")
 	}
 	m := s.init()
-	pattern := regexMap["string_[^\r\n\u2028\u2029]*"]
+	pattern := regexMap["string_.*"]
 	if err := d.ObjBytes(func(d *jx.Decoder, k []byte) error {
 		switch match, err := pattern.Match(k); {
 		case err != nil:

--- a/internal/integration/sample_api_no_otel/oas_schemas_gen.go
+++ b/internal/integration/sample_api_no_otel/oas_schemas_gen.go
@@ -1311,7 +1311,7 @@ func NewIssue943MapIssue943(v Issue943Map) Issue943 {
 // Ref: #/components/schemas/Issue943Map
 type Issue943Map struct {
 	Selector string `json:"selector"`
-	// Pattern: "^variant3_[^\r\n\u2028\u2029]*".
+	// Pattern: "^variant3_.*".
 	Pattern0Props Issue943MapPattern0
 }
 
@@ -5544,7 +5544,7 @@ func (s *RecursiveMapAdditional) init() RecursiveMapAdditional {
 // Ref: #/components/schemas/StringIntMap
 type StringIntMap struct {
 	AdditionalProps StringIntMapAdditional
-	// Pattern: "string_[^\r\n\u2028\u2029]*".
+	// Pattern: "string_.*".
 	Pattern0Props StringIntMapPattern0
 }
 

--- a/internal/integration/sample_api_ns/oas_cfg_gen.go
+++ b/internal/integration/sample_api_ns/oas_cfg_gen.go
@@ -17,10 +17,10 @@ import (
 )
 
 var regexMap = map[string]ogenregex.Regexp{
-	"^\\d-\\d$":                      ogenregex.MustCompile("^\\d-\\d$"),
-	"^variant3_[^\r\n\u2028\u2029]*": ogenregex.MustCompile("^variant3_[^\r\n\u2028\u2029]*"),
-	"foo[^\r\n\u2028\u2029]*":        ogenregex.MustCompile("foo[^\r\n\u2028\u2029]*"),
-	"string_[^\r\n\u2028\u2029]*":    ogenregex.MustCompile("string_[^\r\n\u2028\u2029]*"),
+	"^\\d-\\d$":    ogenregex.MustCompile("^\\d-\\d$"),
+	"^variant3_.*": ogenregex.MustCompile("^variant3_.*"),
+	"foo.*":        ogenregex.MustCompile("foo.*"),
+	"string_.*":    ogenregex.MustCompile("string_.*"),
 }
 var ratMap = map[string]*big.Rat{
 	"10": func() *big.Rat {

--- a/internal/integration/sample_api_ns/oas_json_gen.go
+++ b/internal/integration/sample_api_ns/oas_json_gen.go
@@ -2942,7 +2942,7 @@ func (s *Issue943Map) Decode(d *jx.Decoder) error {
 			}
 		default:
 			var handled bool
-			switch match, err := regexMap["^variant3_[^\r\n\u2028\u2029]*"].Match(k); {
+			switch match, err := regexMap["^variant3_.*"].Match(k); {
 			case err != nil:
 				return errors.Wrap(err, "execute regex")
 			case match:
@@ -3040,7 +3040,7 @@ func (s *Issue943MapPattern0) Decode(d *jx.Decoder) error {
 		return errors.New("invalid: unable to decode Issue943MapPattern0 to nil")
 	}
 	m := s.init()
-	pattern := regexMap["^variant3_[^\r\n\u2028\u2029]*"]
+	pattern := regexMap["^variant3_.*"]
 	if err := d.ObjBytes(func(d *jx.Decoder, k []byte) error {
 		switch match, err := pattern.Match(k); {
 		case err != nil:
@@ -6092,7 +6092,7 @@ func (s *OnlyPatternedPropsObject) Decode(d *jx.Decoder) error {
 		return errors.New("invalid: unable to decode OnlyPatternedPropsObject to nil")
 	}
 	m := s.init()
-	pattern := regexMap["string_[^\r\n\u2028\u2029]*"]
+	pattern := regexMap["string_.*"]
 	if err := d.ObjBytes(func(d *jx.Decoder, k []byte) error {
 		switch match, err := pattern.Match(k); {
 		case err != nil:
@@ -7625,7 +7625,7 @@ func (s *PatternRecursiveMap) Decode(d *jx.Decoder) error {
 		return errors.New("invalid: unable to decode PatternRecursiveMap to nil")
 	}
 	m := s.init()
-	pattern := regexMap["foo[^\r\n\u2028\u2029]*"]
+	pattern := regexMap["foo.*"]
 	if err := d.ObjBytes(func(d *jx.Decoder, k []byte) error {
 		switch match, err := pattern.Match(k); {
 		case err != nil:
@@ -8742,7 +8742,7 @@ func (s *StringIntMap) Decode(d *jx.Decoder) error {
 		switch string(k) {
 		default:
 			var handled bool
-			switch match, err := regexMap["string_[^\r\n\u2028\u2029]*"].Match(k); {
+			switch match, err := regexMap["string_.*"].Match(k); {
 			case err != nil:
 				return errors.Wrap(err, "execute regex")
 			case match:
@@ -8875,7 +8875,7 @@ func (s *StringIntMapPattern0) Decode(d *jx.Decoder) error {
 		return errors.New("invalid: unable to decode StringIntMapPattern0 to nil")
 	}
 	m := s.init()
-	pattern := regexMap["string_[^\r\n\u2028\u2029]*"]
+	pattern := regexMap["string_.*"]
 	if err := d.ObjBytes(func(d *jx.Decoder, k []byte) error {
 		switch match, err := pattern.Match(k); {
 		case err != nil:

--- a/internal/integration/sample_api_ns/oas_schemas_gen.go
+++ b/internal/integration/sample_api_ns/oas_schemas_gen.go
@@ -1311,7 +1311,7 @@ func NewIssue943MapIssue943(v Issue943Map) Issue943 {
 // Ref: #/components/schemas/Issue943Map
 type Issue943Map struct {
 	Selector string `json:"selector"`
-	// Pattern: "^variant3_[^\r\n\u2028\u2029]*".
+	// Pattern: "^variant3_.*".
 	Pattern0Props Issue943MapPattern0
 }
 
@@ -5544,7 +5544,7 @@ func (s *RecursiveMapAdditional) init() RecursiveMapAdditional {
 // Ref: #/components/schemas/StringIntMap
 type StringIntMap struct {
 	AdditionalProps StringIntMapAdditional
-	// Pattern: "string_[^\r\n\u2028\u2029]*".
+	// Pattern: "string_.*".
 	Pattern0Props StringIntMapPattern0
 }
 

--- a/internal/integration/sample_api_nsnc/oas_cfg_gen.go
+++ b/internal/integration/sample_api_nsnc/oas_cfg_gen.go
@@ -10,10 +10,10 @@ import (
 )
 
 var regexMap = map[string]ogenregex.Regexp{
-	"^\\d-\\d$":                      ogenregex.MustCompile("^\\d-\\d$"),
-	"^variant3_[^\r\n\u2028\u2029]*": ogenregex.MustCompile("^variant3_[^\r\n\u2028\u2029]*"),
-	"foo[^\r\n\u2028\u2029]*":        ogenregex.MustCompile("foo[^\r\n\u2028\u2029]*"),
-	"string_[^\r\n\u2028\u2029]*":    ogenregex.MustCompile("string_[^\r\n\u2028\u2029]*"),
+	"^\\d-\\d$":    ogenregex.MustCompile("^\\d-\\d$"),
+	"^variant3_.*": ogenregex.MustCompile("^variant3_.*"),
+	"foo.*":        ogenregex.MustCompile("foo.*"),
+	"string_.*":    ogenregex.MustCompile("string_.*"),
 }
 var ratMap = map[string]*big.Rat{
 	"10": func() *big.Rat {

--- a/internal/integration/sample_api_nsnc/oas_json_gen.go
+++ b/internal/integration/sample_api_nsnc/oas_json_gen.go
@@ -2942,7 +2942,7 @@ func (s *Issue943Map) Decode(d *jx.Decoder) error {
 			}
 		default:
 			var handled bool
-			switch match, err := regexMap["^variant3_[^\r\n\u2028\u2029]*"].Match(k); {
+			switch match, err := regexMap["^variant3_.*"].Match(k); {
 			case err != nil:
 				return errors.Wrap(err, "execute regex")
 			case match:
@@ -3040,7 +3040,7 @@ func (s *Issue943MapPattern0) Decode(d *jx.Decoder) error {
 		return errors.New("invalid: unable to decode Issue943MapPattern0 to nil")
 	}
 	m := s.init()
-	pattern := regexMap["^variant3_[^\r\n\u2028\u2029]*"]
+	pattern := regexMap["^variant3_.*"]
 	if err := d.ObjBytes(func(d *jx.Decoder, k []byte) error {
 		switch match, err := pattern.Match(k); {
 		case err != nil:
@@ -6092,7 +6092,7 @@ func (s *OnlyPatternedPropsObject) Decode(d *jx.Decoder) error {
 		return errors.New("invalid: unable to decode OnlyPatternedPropsObject to nil")
 	}
 	m := s.init()
-	pattern := regexMap["string_[^\r\n\u2028\u2029]*"]
+	pattern := regexMap["string_.*"]
 	if err := d.ObjBytes(func(d *jx.Decoder, k []byte) error {
 		switch match, err := pattern.Match(k); {
 		case err != nil:
@@ -7625,7 +7625,7 @@ func (s *PatternRecursiveMap) Decode(d *jx.Decoder) error {
 		return errors.New("invalid: unable to decode PatternRecursiveMap to nil")
 	}
 	m := s.init()
-	pattern := regexMap["foo[^\r\n\u2028\u2029]*"]
+	pattern := regexMap["foo.*"]
 	if err := d.ObjBytes(func(d *jx.Decoder, k []byte) error {
 		switch match, err := pattern.Match(k); {
 		case err != nil:
@@ -8742,7 +8742,7 @@ func (s *StringIntMap) Decode(d *jx.Decoder) error {
 		switch string(k) {
 		default:
 			var handled bool
-			switch match, err := regexMap["string_[^\r\n\u2028\u2029]*"].Match(k); {
+			switch match, err := regexMap["string_.*"].Match(k); {
 			case err != nil:
 				return errors.Wrap(err, "execute regex")
 			case match:
@@ -8875,7 +8875,7 @@ func (s *StringIntMapPattern0) Decode(d *jx.Decoder) error {
 		return errors.New("invalid: unable to decode StringIntMapPattern0 to nil")
 	}
 	m := s.init()
-	pattern := regexMap["string_[^\r\n\u2028\u2029]*"]
+	pattern := regexMap["string_.*"]
 	if err := d.ObjBytes(func(d *jx.Decoder, k []byte) error {
 		switch match, err := pattern.Match(k); {
 		case err != nil:

--- a/internal/integration/sample_api_nsnc/oas_schemas_gen.go
+++ b/internal/integration/sample_api_nsnc/oas_schemas_gen.go
@@ -1311,7 +1311,7 @@ func NewIssue943MapIssue943(v Issue943Map) Issue943 {
 // Ref: #/components/schemas/Issue943Map
 type Issue943Map struct {
 	Selector string `json:"selector"`
-	// Pattern: "^variant3_[^\r\n\u2028\u2029]*".
+	// Pattern: "^variant3_.*".
 	Pattern0Props Issue943MapPattern0
 }
 
@@ -5544,7 +5544,7 @@ func (s *RecursiveMapAdditional) init() RecursiveMapAdditional {
 // Ref: #/components/schemas/StringIntMap
 type StringIntMap struct {
 	AdditionalProps StringIntMapAdditional
-	// Pattern: "string_[^\r\n\u2028\u2029]*".
+	// Pattern: "string_.*".
 	Pattern0Props StringIntMapPattern0
 }
 

--- a/ogenregex/ogenregex.go
+++ b/ogenregex/ogenregex.go
@@ -22,7 +22,8 @@ var _ = []Regexp{
 }
 
 type goRegexp struct {
-	exp *regexp.Regexp
+	orig string
+	exp  *regexp.Regexp
 }
 
 func (r goRegexp) Match(s []byte) (bool, error) {
@@ -34,7 +35,7 @@ func (r goRegexp) MatchString(s string) (bool, error) {
 }
 
 func (r goRegexp) String() string {
-	return r.exp.String()
+	return r.orig
 }
 
 type regexp2Regexp struct {
@@ -67,7 +68,7 @@ type Regexp interface {
 func Compile(exp string) (Regexp, error) {
 	if converted, ok := Convert(exp); ok {
 		if re, err := regexp.Compile(converted); err == nil {
-			return goRegexp{re}, nil
+			return goRegexp{orig: exp, exp: re}, nil
 		}
 	}
 	re, err := regexp2.Compile(exp, regexp2.ECMAScript|regexp2.Unicode)

--- a/ogenregex/ogenregex_test.go
+++ b/ogenregex/ogenregex_test.go
@@ -9,68 +9,68 @@ import (
 
 func TestCompile(t *testing.T) {
 	type testCase struct {
-		input      string
-		wantType   Regexp
-		wantString string
-		wantErr    bool
+		input    string
+		wantType Regexp
+		wantErr  bool
 	}
 
 	tests := []testCase{
 		// Conversion is not required.
-		{`\x20`, goRegexp{}, `\x20`, false},
-		{`\v`, goRegexp{}, `\v`, false},
-		{`\t`, goRegexp{}, `\t`, false},
-		{`\n`, goRegexp{}, `\n`, false},
-		{`\d`, goRegexp{}, `\d`, false},
-		{`\w`, goRegexp{}, `\w`, false},
-		{`\w{1}`, goRegexp{}, `\w{1}`, false},
-		{`\w{1,}`, goRegexp{}, `\w{1,}`, false},
-		{`\w{1,2}`, goRegexp{}, `\w{1,2}`, false},
-		{`\b`, goRegexp{}, `\b`, false},
-		{`\B`, goRegexp{}, `\B`, false},
-		{`\.`, goRegexp{}, `\.`, false},
-		{`\[`, goRegexp{}, `\[`, false},
-		{`\]`, goRegexp{}, `\]`, false},
-		{`\(`, goRegexp{}, `\(`, false},
-		{`\)`, goRegexp{}, `\)`, false},
-		{`\{`, goRegexp{}, `\{`, false},
-		{`\}`, goRegexp{}, `\}`, false},
-		{`\\`, goRegexp{}, `\\`, false},
-		{`\$`, goRegexp{}, `\$`, false},
+		{`\0`, goRegexp{}, false},
+		{`\x20`, goRegexp{}, false},
+		{`\v`, goRegexp{}, false},
+		{`\t`, goRegexp{}, false},
+		{`\n`, goRegexp{}, false},
+		{`\d`, goRegexp{}, false},
+		{`\w`, goRegexp{}, false},
+		{`\w{1}`, goRegexp{}, false},
+		{`\w{1,}`, goRegexp{}, false},
+		{`\w{1,2}`, goRegexp{}, false},
+		{`\b`, goRegexp{}, false},
+		{`\B`, goRegexp{}, false},
+		{`\.`, goRegexp{}, false},
+		{`\[`, goRegexp{}, false},
+		{`\]`, goRegexp{}, false},
+		{`\(`, goRegexp{}, false},
+		{`\)`, goRegexp{}, false},
+		{`\{`, goRegexp{}, false},
+		{`\}`, goRegexp{}, false},
+		{`\\`, goRegexp{}, false},
+		{`\$`, goRegexp{}, false},
 
 		// Simplification.
-		{`\u000a`, goRegexp{}, `\x{000a}`, false},
-		{`\u{000a}`, goRegexp{}, `\x{000a}`, false},
+		{`\u000a`, goRegexp{}, false},
+		{`\u{000a}`, goRegexp{}, false},
 		// "\z" just unnecessarily escapes the 'z'.
-		{`\z`, goRegexp{}, `z`, false},
+		{`\z`, goRegexp{}, false},
 
 		// Conversion is required.
 		//
 		// See https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Regular_Expressions/Character_Classes#types.
 		//
 		// In ECMA-262, \c + [a-fA-F] is a control character.
-		{`\ca`, goRegexp{}, `\x01`, false},
-		{`\cA`, goRegexp{}, `\x01`, false},
-		{`\cb`, goRegexp{}, `\x02`, false},
-		{`\cB`, goRegexp{}, `\x02`, false},
+		{`\ca`, goRegexp{}, false},
+		{`\cA`, goRegexp{}, false},
+		{`\cb`, goRegexp{}, false},
+		{`\cB`, goRegexp{}, false},
 		// In ECMA-262, \b in a character class is a backspace.
-		{`[\b]`, goRegexp{}, `[\x08]`, false},
+		{`[\b]`, goRegexp{}, false},
 		// ECMA-262 dot matches any single character except line terminators: \n, \r, \u2028 or \u2029.
-		{`.*`, goRegexp{}, re2Dot + `*`, false},
+		{`.*`, goRegexp{}, false},
 		// Whitespace characters in ECMA-262 differ from those in RE2.
 		//
 		// Whitespace characters in ECMA-262:
 		// [ \f\n\r\t\v\u00a0\u1680\u2000-\u200a\u2028\u2029\u202f\u205f\u3000\ufeff]
-		{`\s`, goRegexp{}, `[` + whitespaceChars + `]`, false},
-		{`\S`, goRegexp{}, `[^` + whitespaceChars + `]`, false},
-		{`[\s]`, goRegexp{}, `[` + whitespaceChars + `]`, false},
+		{`\s`, goRegexp{}, false},
+		{`\S`, goRegexp{}, false},
+		{`[\s]`, goRegexp{}, false},
 
 		// Use regexp2.
-		{`^(?!examples/)`, regexp2Regexp{}, `^(?!examples/)`, false},
+		{`^(?!examples/)`, regexp2Regexp{}, false},
 
 		// Error.
-		{")", nil, ``, true},
-		{"(?`)", nil, ``, true},
+		{")", nil, true},
+		{"(?`)", nil, true},
 	}
 	for i, tt := range tests {
 		tt := tt
@@ -87,7 +87,84 @@ func TestCompile(t *testing.T) {
 			a.NoError(err)
 			a.NotPanics(func() { MustCompile(tt.input) })
 			a.IsType(tt.wantType, got)
-			a.Equal(tt.wantString, got.String())
+			a.Equal(tt.input, got.String())
+		})
+	}
+}
+
+func TestConvert(t *testing.T) {
+	type testCase struct {
+		input      string
+		wantString string
+		wantOk     bool
+	}
+
+	tests := []testCase{
+		// Conversion is not required.
+		{`\0`, `\0`, true},
+		{`\x20`, `\x20`, true},
+		{`\v`, `\v`, true},
+		{`\t`, `\t`, true},
+		{`\n`, `\n`, true},
+		{`\d`, `\d`, true},
+		{`\w`, `\w`, true},
+		{`\w{1}`, `\w{1}`, true},
+		{`\w{1,}`, `\w{1,}`, true},
+		{`\w{1,2}`, `\w{1,2}`, true},
+		{`\b`, `\b`, true},
+		{`\B`, `\B`, true},
+		{`\.`, `\.`, true},
+		{`\[`, `\[`, true},
+		{`\]`, `\]`, true},
+		{`\(`, `\(`, true},
+		{`\)`, `\)`, true},
+		{`\{`, `\{`, true},
+		{`\}`, `\}`, true},
+		{`\\`, `\\`, true},
+		{`\$`, `\$`, true},
+
+		// Simplification.
+		{`\u000a`, `\x{000a}`, true},
+		{`\u{000a}`, `\x{000a}`, true},
+		// "\z" just unnecessarily escapes the 'z'.
+		{`\z`, `z`, true},
+
+		// Conversion is required.
+		//
+		// See https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Regular_Expressions/Character_Classes#types.
+		//
+		// In ECMA-262, \c + [a-fA-F] is a control character.
+		{`\ca`, `\x01`, true},
+		{`\cA`, `\x01`, true},
+		{`\cb`, `\x02`, true},
+		{`\cB`, `\x02`, true},
+		// In ECMA-262, \b in a character class is a backspace.
+		{`[\b]`, `[\x08]`, true},
+		// ECMA-262 dot matches any single character except line terminators: \n, \r, \u2028 or \u2029.
+		{`.*`, re2Dot + `*`, true},
+		// Whitespace characters in ECMA-262 differ from those in RE2.
+		//
+		// Whitespace characters in ECMA-262:
+		// [ \f\n\r\t\v\u00a0\u1680\u2000-\u200a\u2028\u2029\u202f\u205f\u3000\ufeff]
+		{`\s`, `[` + whitespaceChars + `]`, true},
+		{`\S`, `[^` + whitespaceChars + `]`, true},
+		{`[\s]`, `[` + whitespaceChars + `]`, true},
+
+		// Use regexp2.
+		{`^(?!examples/)`, ``, false},
+
+		// Error.
+		{")", ``, false},
+		{"(?`)", ``, false},
+	}
+	for i, tt := range tests {
+		tt := tt
+		t.Run(fmt.Sprintf("Test%d", i+1), func(t *testing.T) {
+			a := require.New(t)
+
+			got, ok := Convert(tt.input)
+			a.Equal(tt.wantOk, ok, "%q", tt.input)
+			a.Equal(tt.wantString, got)
 		})
 	}
 }

